awx/main/scheduler/task_manager_models.py

```diff
@@ -121,3 +123,16 @@ def find_largest_idle_instance(self, instance_group_name, capacity_type='executi
                 elif i.capacity > largest_instance.capacity:
                     largest_instance = i
         return largest_instance
+
+    def get_instance_groups_from_task_cache(self, task):
+        igs = []
+        for pk in task.preferred_instance_groups_cache:
```

Comment:
Imagine that you have a job from before the migration. In that case `task.preferred_instance_groups_cache` may be `None` because of the default you set. The _last_ thing we want is an uncaught exception here, so I would advocate for further, aggressive, data sanitization. For another pathway, someone could create a job with `awx-manage shell_plus`, and you don't want to grind to a halt because it didn't take the canonical creation path. In all of these cases, there's likely very little care for the specific IG / instance selection. A job went rogue, and the user just wants you to move it along and let the work happen.